### PR TITLE
Enforce site filter for Groups restore

### PR DESCRIPTION
**Pull request by meain**
This PR reworks the groups restore to and makes the site to restore to mandatory. This also updates some missing filtering capabilities in groups export.

---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* closes https://github.com/alcionai/corso/issues/4462

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/cli/utils/groups.go [hunk 1]
```diff
@@ -101,11 +101,18 @@ func MakeGroupsOpts(cmd *cobra.Command) GroupsOpts {
...
+func ValidateGroupsRestoreFlags(backupID string, opts GroupsOpts, isRestore bool) error {
 	if len(backupID) == 0 {
 		return clues.New("a backup ID is required")
 	}
 
+	// The user has to explicitly specify which resource to restore. In
+	// this case, since we can only restore sites, the user is supposed
+	// to specify which site to restore.
+	if isRestore && len(opts.Site) == 0 {
+		return clues.New("a site ID or web URL is required")
```
Comment (ryanfkeepers) [inline src/cli/utils/groups.go:113]: Our standard in sharepoint is to separate siteIDs from weburls with the --site-id and --site flags, respectively. Reducing it to --site works, but i'm a bit worried about confusing end users.  Mostly speaking out loud.  We should probably find a way to remove the --site-id flag for sharepoint to keep things uniform across service commands.

**Review by ryanfkeepers — APPROVED** [review]
nits about flag duplication, but approving to unblock

File: src/cli/flags/groups.go [hunk 2]
```diff
@@ -22,12 +23,32 @@ var (
 	GroupFV   []string
 	MessageFV []string
 
+	// Have to create a separate one for restore to make sure we
+	// create on which accepts only one value
+	GroupSiteFV string
```
[the remark this answers is not in the document]
Comment (meain) [reply src/cli/flags/groups.go:28]: I guess that should work. The only limitation seems to be that we cannot have a "required" flag at the cobra level as we not have two possible options for the user to use.

File: src/cli/utils/groups.go (hunk 1 above)
Comment (meain) [reply src/cli/utils/groups.go:113]: Yeah, I think we could have just had a single flag. I don't wanna change that behaviour in this PR. I think that should be a separate larger change.

**Pull request merged**